autoscheduler/frontend/src/tests/ui/SchedulingPage.test.tsx

```diff
@@ -10,11 +10,67 @@ import { Provider } from 'react-redux';
 import {
   render, fireEvent, waitFor, queryByText as queryContainerByText,
 } from '@testing-library/react';
+import * as router from '@reach/router';
 import autoSchedulerReducer from '../../redux/reducer';
 import SchedulingPage from '../../components/SchedulingPage/SchedulingPage';
 import { mockFetchSchedulerGenerate } from '../testData';
 
 describe('Scheduling Page UI', () => {
+  // setup and teardown spy function on navigate
+  const navSpy = jest.spyOn(router, 'navigate');
+  // don't actually call navigate
+  beforeAll(() => navSpy.mockImplementation(() => null));
+  // restore navigate to original
+  afterAll(navSpy.mockRestore);
```

Comment:
What's your thought-process for this being `beforeAll`/`afterAll` instead of `beforeEach`/`afterEach`?

Reply:
So basically the `beforeAll` changes the navigate function to not do anything. The other tests in the file fail if navigate is actually called since navigating when app is null causes undefined behavior (I think). So you don't actually need to re-do that multiple times. The counter is reset by `beforeAll(navSpy.mockClear)` once in the beginning of the doesn't redirect test, otherwise the counter doesn't matter anyways so I tried to clear/reset as little as possible for speed.